persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -231,13 +237,18 @@ public Authenticator getAuthenticator()
                 }
 
                 CQLStatement statement = QueryProcessor.parseStatement(cql, internalState).statement;
-                if (!StargateSystemKeyspace.maybeCompleteSystemLocalOrPeers(statement, internalState, internalOptions, queryStartNanoTime, future))
+
+                Result result;
+                if  (interceptor.shouldInterceptQuery(statement, state, options, customPayload, queryStartNanoTime))
+                    result = interceptor.interceptQuery(handler, statement, state, options, customPayload, queryStartNanoTime);
```

Comment:
This could be simplified to a single call, e.g.:

```java
result = interceptor.interceptQuery(...);

if (result == null)
{
  // normal path
}
```

Reply:
Good suggestion.